core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -50,14 +52,20 @@ class OffsetCheckpoint(val file: File) extends Logging {
           writer.write("%s %d %d".format(topicPart.topic, topicPart.partition, offset))
           writer.newLine()
         }
-      
+
         // flush the buffer and then fsync the underlying file
         writer.flush()
         fileOutputStream.getFD().sync()
+      } catch {
+        case e: FileNotFoundException =>
+          if (FileSystems.getDefault.isReadOnly) {
+            fatal("Halting writes to offset checkpoint file because the underlying file system is inaccessible : ", e)
+            Runtime.getRuntime.halt(1)
+          }
```

Comment:
@ijuma 
By conflicts, you mean the comments right ?

We should rethrow the exception in case the if statement evaluates to false, right?
---> Sure. Will upload a new patch.

I agree that we should have the halt() call at one place. Right now there are multiple places where this is happening in code and as explained in one of the comments above write() is accessed at multiple places and some of them catch IOException (Not KafkaStorageException) to do the runtime.halt(), for example checkpointHighWatermarks() in ReplicaManger.scala. Also calls to OffsetCheckpoint.write() may be async. 
If w have to do this, we have to do it all places to be consistent and would require a bigger patch then this. What do you think?